cpp/src/arrow/array/builder_binary.cc

```diff
@@ -232,8 +232,8 @@ Status FixedSizeBinaryBuilder::AppendValues(const uint8_t* data, int64_t length,
 
 Status FixedSizeBinaryBuilder::AppendNull() {
   RETURN_NOT_OK(Reserve(1));
-  UnsafeAppendToBitmap(false);
-  return byte_builder_.Advance(byte_width_);
+  UnsafeAppendNull();
+  return Status::OK();
```

Comment:
Hmm... can you explain why you removed the `Advance()` call?

Reply:
`UnsafeAppendNull()` is a function in `FixedSizeBinaryBuilder`; [Advance still gets called.](https://github.com/apache/arrow/pull/3880/files/3a1ef1285e591d71f6aca125c0662e6cc6c4eab0#diff-011119a62ed7b6dcdc29e9e46674ee94R229) I refactored to satisfy the pattern that AppendX = ReserveX, UnsafeAppendX, return OK

Reply:
Ah, thank you.